### PR TITLE
Add Metsim Recipe

### PR DESCRIPTION
**Pull request by jhamman**
cc @bartnijssen, @arbennett

closes https://github.com/UW-Hydro/MetSim/issues/80

### CODE REVIEW

File: recipes/metsim/meta.yaml [hunk 1]
```diff
@@ -0,0 +1,56 @@
...
+  name: {{ name|lower }}
+  version: {{ version }}
+
+source:
+  fn: {{ name }}-{{ version }}.tar.gz
+  url: https://github.com/UW-Hydro/{{ name }}/archive/v{{ version }}.tar.gz
+  sha256: {{ sha256 }}
+
+build:
+  number: 0
```
Comment (CJ-Wright) [inline recipes/metsim/meta.yaml]: Could this be noarch?

File: recipes/metsim/meta.yaml [hunk 2]
```diff
@@ -0,0 +1,56 @@
...
+  script: python setup.py install --single-version-externally-managed --record record.txt
+requirements:
```
Comment (CJ-Wright) [inline recipes/metsim/meta.yaml]: Please put a blank line between build and requirements

**Comment by jhamman** [comment]
Thanks @CJ-Wright, comments addressed. Question for you, I'm getting a `raise Exception('info/files')` error on windows and osx. Any ideas?

**Comment by CJ-Wright** [comment]
It seems that something is taking exception to the scripts in the `scripts` folder. I haven't seen this kind of error before (I usually ship things as entry points).  

You might try reverting from pip to `python setup.py install --single-version-externally-managed --record record.txt` for the install.

@conda-forge/staged-recipes thoughts?

**Comment by jhamman** [comment]
No beans with `setup.py`. 

 @arbennett, any ideas?

**Comment by ocefpaf** [comment]
> It seems that something is taking exception to the scripts in the scripts folder. I haven't seen this kind of error before (I usually ship things as entry points).

Indeed. We need to patch it to be `entry_points`, scripts won't work until they fix conda-build upstream.

**Comment by jhamman** [comment]
Thanks @ocefpaf.

We'll issue another release soon with an updated setup.py. Until then, this issue is on hold.

**Comment by jhamman** [comment]
cc @arbennett

File: recipes/metsim/meta.yaml [hunk 3]
```diff
@@ -39,7 +39,6 @@ about:
   home: https://github.com/UW-Hydro/MetSim
   license: GPL-3.0
   license_family: GPL3
-  license_file: LICENSE
```
Comment (ocefpaf) [inline recipes/metsim/meta.yaml]: B/c the license is not in the tarball, you'll need a `MANIFEST.in` upstream, you'll had to:

https://conda-forge.org/docs/meta.html#packaging-the-license-manually

File: recipes/metsim/meta.yaml [hunk 4]
```diff
@@ -0,0 +1,60 @@
...
+source:
+  url: https://pypi.io/packages/source/m/metsim/metsim-{{ version }}.tar.gz
+  sha256: 96598273aee7fd0d502699395f0ffaa3e4a780dfbdf28c35c1c541d2985f9154
+
+build:
+  number: 0
+  noarch: python
+  script: python -m pip install --no-deps --ignore-installed .
+  entry_points:
+    - ms = metsim.cli.ms:main
```
Comment (ocefpaf) [inline recipes/metsim/meta.yaml]: You'll need this for the `commands` test to pass.

File: recipes/metsim/meta.yaml [hunk 5]
```diff
@@ -0,0 +1,60 @@
...
+    - metsim.cli
+  commands:
+    - ms --help
+
+about:
+  home: https://github.com/UW-Hydro/MetSim
+  license: GPL-3.0
+  license_family: GPL3
+  # TODO: remove license_file after https://github.com/UW-Hydro/MetSim/pull/138
+  license_file: '{{ environ["RECIPE_DIR"] }}/LICENSE'
```
Comment (ocefpaf) [inline recipes/metsim/meta.yaml]: Local licenses need the full path.

File: recipes/metsim/meta.yaml [hunk 6]
```diff
@@ -1,37 +1,40 @@
...
 {% set version = "1.1.0" %}
-{% set sha256 = "96598273aee7fd0d502699395f0ffaa3e4a780dfbdf28c35c1c541d2985f9154" %}
 
 package:
-  name: {{ name|lower }}
+  name: metsim
   version: {{ version }}
 
 source:
-  fn: {{ name }}-{{ version }}.tar.gz
```
Comment (ocefpaf) [inline recipes/metsim/meta.yaml]: Just some simplification to make the recipe more readable.

**Comment by ocefpaf** [comment]
This is `noarch` so we don't need to wait for Travis-CI.

**Comment by jhamman** [comment]
@ocefpaf - you are my hero.

**Pull request merged**